helm/cluster-eks/values.schema.json

```diff
@@ -396,6 +396,31 @@
                             ],
                             "default": "public"
                         },
+                        "encryptionConfig": {
+                            "type": "object",
+                            "title": "Encryption config",
+                            "description": "Encryption configuration for the Kubernetes API server.",
+                            "properties": {
+                                "keyARN": {
+                                    "type": "string",
+                                    "title": "Key ARN",
+                                    "description": "AWS KMS key ARN. To enable encryption when creating a cluster you need to create a new KMS key that has an alias name starting with cluster-api-provider-aws-. For example, arn:aws:kms:eu-north-1:12345678901:alias/cluster-api-provider-aws-key1.",
+                                    "examples": [
+                                        "arn:aws:kms:eu-west-1:123456789012:key/1234abcd-12ab-34cd-56ef-1234567890ab"
+                                    ]
+                                },
+                                "resources": {
+                                    "type": "array",
+                                    "title": "Resources",
+                                    "description": "AWS resources to encrypt. Default to secrets if not specified.",
+                                    "items": {
+                                        "type": "string",
+                                        "title": "Resource",
+                                        "description": "AWS resource to encrypt, for example `secrets`."
+                                    }
```

Comment:
Can we specify what the items mean? Are they CRD names/aliases as in `kubectl api-resources` output?

Reply:
This is more of a copy from the CRD definition of the AWSManagedControlPlane, I don't think there is a possibility to encrypt anything else than a secret in the k8s datastore so far but I guess they have added it there for potential future changes.

Reply:
and cluster-api-provider-aws controller based this on the AWS API
```
Specifies the resources to be encrypted. The only supported value is secrets .
```

So atm AWS only supports secrets  and there is no specification on how would something else look like